core/src/event.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::db::Key;
+use coordinator::types::Event;
+use ctypes::{BlockHash, TxHash};
+use primitives::H256;
+use rlp::{Decodable, Encodable, Rlp, RlpStream};
+use std::hash::Hash;
+use std::ops::Deref;
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub enum EventSource {
+    Block(BlockHash),
+    Transaction(TxHash),
+}
+
+#[derive(Clone, Default, PartialEq, Eq)]
+pub struct Events(pub Vec<Event>);
+
+impl Encodable for Events {
+    fn rlp_append(&self, _: &mut RlpStream) {
+        unimplemented!()
```

Comment:
Actually, no encoding scheme is decided yet, so it's okay to keep it unimplemented. 

However, I already implemented RLP encoding/decoding traits for other coordinator-related types in order to test some behavior, although the encoding scheme is subject to change.

Since I already implemented RLP for many other types, I think it's not a bad idea to implement it in Events type.
I'll add EventDB tests as well. (The reason I didn't implement RLP encoding for Events was that I didn't implement any EventDB tests)

Reply:
I added Encodable trait to Event and added some EventDB tests.